doc/lsst.pipe.base/command-line-task-prov-howto.rst

```diff
@@ -0,0 +1,76 @@
+.. _command-line-task-prov-howto:
+
+####################################################
+Working with provenance checks in command-line tasks
+####################################################
+
+:ref:`Command-line tasks <using-command-line-tasks>` include provenance checks to ensure datasets in a Butler repository are processed with consistent software versions and task configurations.
+This page provides background on what these provenance checks are, and how they can be overridden when running command-line tasks.
+
+.. _command-line-task-prov-howto-config:
+
+How to override configuration checks with the :option:`--clobber-config` argument
+=================================================================================
+
+When a command-line task outputs datasets it also persists the task's configuration to the same Butler output repository.
+If the same command-line task is later run on the same Butler output repository, that task will ensure that the configuration is consistent with the earlier run.
+If the current and previous task configurations are different, the command-line task will abort with an error:
+
+.. code-block:: text
+
+   Config does not match existing task config <name> on disk; tasks configurations must be consistent within the same output repo (override with --clobber-config)
+
+This check ensures that datasets in a Butler repository are uniformly processed.
+
+You may often experiment with different task configurations while prototyping a pipeline.
+The best way to avoid configuration conflicts is to run each experimental command-line task execution with a different output :ref:`rerun <command-line-task-data-repo-using-reruns>`.
+Creating separate reruns gives you the ability to later analyze how dataset differ with task configuration.
+See :ref:`command-line-task-data-repo-using-reruns` for details on how to do this.
+
+If you cannot create a new output rerun, and must re-use an existing data repository, you can overwrite (clobber) existing task configuration metadata with the current configurations to bypass the consistency check.
+Use the :option:`--clobber-config` argument to do so:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id --clobber-config ...
+
+.. warning::
+
+   :option:`--clobber-config` is subject to race conditions because it is modifying a file in the Butler repository.
+   It should not be used in conjunction with parallel processing (such as :option:`-j`/:option:`--processes` or with "driver" command-line tasks).
+
```

Comment:
As noted elsewhere these docs are inconsistent on whether `--clobber-config` is safe with `-j`